12-hooks-in-react/src/components/Dropdown.js

```diff
@@ -1,7 +1,28 @@
-import React, { useState } from "react";
+import React, { useState, useEffect, useRef } from "react";
 
 const Dropdown = ({ options, selected, onSelectedChange }) => {
   const [open, setOpen] = useState(false);
+  const ref = useRef();
+
+  useEffect(() => {
+    const onBodyClick = (event) => {
+      // If user click in the dropdown, do nothing in this function.
+      if (ref.current.contains(event.target)) return;
```

Comment:
Good use of guard clause, remember this: return early

Reply:
Thanks! :100: